var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -286,6 +286,21 @@ class Trilinos(CMakePackage, CudaPackage):
     # Fortran mangling fails on Apple M1 (see spack/spack#25900)
     conflicts('@:13.0.1 +fortran', when='target=m1')
 
+    amdgpu_arch_map = {
+        'gfx900': 'vega900',
+        'gfx906': 'vega906',
+        'gfx908': 'vega908',
+        'gfx90a': 'vega90A'
+    }
+    amd_support_conflict_msg = (
+        '{0} is not supported; '
+        'Kokkos supports the following AMD GPU targets: '
+        + ', '.join(amdgpu_arch_map.keys()))
+    for arch in ROCmPackage.amdgpu_targets:
+        if arch not in amdgpu_arch_map:
+            conflicts('+rocm', when='amdgpu_target={0}'.format(arch),
+                      msg=amd_support_conflict_msg.format(arch))
```

Comment:
Definitely make `arch` private rather than an unintentional class property; probably prefix the other variables with `_` as well, and use `set` operations to better express your loop:

```suggestion
    for _arch in set(ROCmPackage.amdgpu_targets) - set(amdgpu_arch_map):
        conflicts('+rocm', when='amdgpu_target=' + _arch,
                  msg=amd_support_conflict_msg.format(_arch))
```